teeio-validator/library/pcie_ide_test_lib/test_case/test_case_keyprog_2.c

```diff
@@ -59,58 +56,67 @@ bool test_ide_km_key_prog_case2(const void *pci_doe_context,
+        ret = false;
+        goto PcieIdeKeyProg2Done;
     }
 
     // Assertion.1
     res = response_size == sizeof(pci_ide_km_kp_ack_t);
-    TEEIO_PRINT(("         %s.1: %s %s\n", case_msg, mKeyProgAssersion[1], res ? "Pass" : "Fail"));
+    assertion_result = res ? IDE_COMMON_TEST_CASE_ASSERTION_RESULT_SUCCESS : IDE_COMMON_TEST_CASE_ASSERTION_RESULT_FAILED;
+    teeio_record_assertion_result(case_class, case_id, 1, IDE_COMMON_TEST_CASE_ASSERTION_TYPE_TEST, assertion_result, "sizeof(IdeKmMessage) = 0x%lx", response_size);
+    left_assertion_cnt--;
     if(!res) {
-        return false;
+        ret = false;
+        goto PcieIdeKeyProg2Done;
     }
 
     // Assertion.2
     res = response.header.object_id == PCI_IDE_KM_OBJECT_ID_KP_ACK;
-    TEEIO_PRINT(("         %s.2: %s %s\n", case_msg, mKeyProgAssersion[2], res ? "Pass" : "Fail"));
-    if(!res) {
-        return false;
-    }
+    assertion_result = res ? IDE_COMMON_TEST_CASE_ASSERTION_RESULT_SUCCESS : IDE_COMMON_TEST_CASE_ASSERTION_RESULT_FAILED;
+    teeio_record_assertion_result(case_class, case_id, 2, IDE_COMMON_TEST_CASE_ASSERTION_TYPE_TEST, assertion_result, "IdeKmMessage.ObjectID = 0x%x", response.header.object_id);
+    left_assertion_cnt--;
+    ret = res && ret;
 
     // Assertion.3
     res = response.status == PCI_IDE_KM_KP_ACK_STATUS_INCORRECT_LENGTH;
-    TEEIO_PRINT(("         %s.3: %s(0x%x) %s\n", case_msg, mKeyProgAssersion[3], status, res ? "Pass" : "Fail"));
-    if(!res) {
-        return false;
-    }
+    assertion_result = res ? IDE_COMMON_TEST_CASE_ASSERTION_RESULT_SUCCESS : IDE_COMMON_TEST_CASE_ASSERTION_RESULT_FAILED;
+    teeio_record_assertion_result(case_class, case_id, 3, IDE_COMMON_TEST_CASE_ASSERTION_TYPE_TEST, assertion_result, "IdeKmMessage.Status = 0x%02x", response.status);
+    left_assertion_cnt--;
+    ret = res && ret;
 
     // Assertion.4
     res = (response.port_index == request.port_index);
-    TEEIO_PRINT(("         %s.3: %s %s\n", case_msg, mKeyProgAssersion[4], res ? "Pass" : "Fail"));
-    if(!res) {
-        return false;
-    }
+    assertion_result = res ? IDE_COMMON_TEST_CASE_ASSERTION_RESULT_SUCCESS : IDE_COMMON_TEST_CASE_ASSERTION_RESULT_FAILED;
+    teeio_record_assertion_result(case_class, case_id, 4, IDE_COMMON_TEST_CASE_ASSERTION_TYPE_TEST, assertion_result, "IdeKmMessage.PortIndex = 0x%x", response.port_index);
+    left_assertion_cnt--;
+    ret = res && ret;
 
     // Assertion.5
     res = (response.stream_id == request.stream_id);
-    TEEIO_PRINT(("         %s.2: %s %s\n", case_msg, mKeyProgAssersion[5], res ? "Pass" : "Fail"));
-    if(!res) {
-        return false;
-    }
+    assertion_result = res ? IDE_COMMON_TEST_CASE_ASSERTION_RESULT_SUCCESS : IDE_COMMON_TEST_CASE_ASSERTION_RESULT_FAILED;
+    teeio_record_assertion_result(case_class, case_id, 5, IDE_COMMON_TEST_CASE_ASSERTION_TYPE_TEST, assertion_result, "IdeKmMessage.StreamID = 0x%x", response.stream_id);
+    left_assertion_cnt--;
+    ret = res && ret;
```

Comment:
It is not really needed, the test code should ways return SUCCESS. That meaning is: test is done, no matter it pass or fail. The final test result should already be recorded via teeio_record_assertion_result().